### PR TITLE
Add Character Assets

### PR DESCRIPTION
**Pull request by Anisetobegan**
(cherry picked from commit c251be7ec5f82714c58fb629b865c9fe8a479b94)

### CODE REVIEW

File: Assets/Scripts/Character.cs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+       
+
+    private void Move()
+    {
+        float xDirection = Input.GetAxis("Horizontal");
+        float zDirection = Input.GetAxis("Vertical");
+
+        Vector3 moveDirection = new Vector3(xDirection, 0.0f, zDirection);
+
+        transform.position += moveDirection * moveSpeed;
```
Comment (morathias) [inline Assets/Scripts/Character.cs]: Multiply by deltatime

File: Assets/Scripts/Character.cs [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+    float moveSpeed = 0.005f;
+    float health;
+    int lvl;
+    float exp;
+    //Perks perk;
+    float summonCooldown;
+    int maxSummons = 5;
+    public List<Summon> currentSummons;
+    bool state = true;
```
Comment (morathias) [inline Assets/Scripts/Character.cs]: state should be of type enum State

File: Assets/Scripts/Character.cs [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+{
+
+    float moveSpeed = 0.005f;
+    float health;
+    int lvl;
+    float exp;
+    //Perks perk;
+    float summonCooldown;
+    int maxSummons = 5;
+    public List<Summon> currentSummons;
```
Comment (morathias) [inline Assets/Scripts/Character.cs]: make it private

File: Assets/Scripts/Character.cs [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+        transform.position += moveDirection * moveSpeed;
+    }
+
+    private void Summon(Summon summon)
+    {
+        if (currentSummons.Count < maxSummons)
+        {
+            Instantiate(summon);
+            currentSummons.Add(summon);
```
Comment (morathias) [inline Assets/Scripts/Character.cs]: You need to add the instantiated object, not the prefab

File: Assets/Scripts/Character.cs [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+        if (Input.GetKeyUp(KeyCode.Alpha2))
+        {
+            Summon(summons[1]);
+        }
+
+        if (Input.GetKeyUp(KeyCode.Alpha3))
+        {
+            Summon(summons[2]);
+        }
```
Comment (morathias) [inline Assets/Scripts/Character.cs:110]: Encapsulate the move and summon code in a switch case (state machine), and this code should be called in the idle and moving states

**Review by morathias — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Character.cs (hunk 2 above)
Comment (Anisetobegan) [reply Assets/Scripts/Character.cs]: Done

File: Assets/Scripts/Character.cs (hunk 3 above)
Comment (Anisetobegan) [reply Assets/Scripts/Character.cs]: Done

File: Assets/Scripts/Character.cs (hunk 4 above)
Comment (Anisetobegan) [reply Assets/Scripts/Character.cs]: Done

File: Assets/Scripts/Character.cs (hunk 1 above)
Comment (Anisetobegan) [reply Assets/Scripts/Character.cs]: Done

File: Assets/Scripts/Character.cs (hunk 5 above)
Comment (Anisetobegan) [reply Assets/Scripts/Character.cs:110]: Done

**Review by morathias — APPROVED** [review]

**Pull request merged**